nni/compression/pytorch/speedup/compressor.py

```diff
@@ -1,143 +1,336 @@
         masks_file : str
             The path of user provided mask file
         map_location : str
             the device on which masks are placed, same to map_location in ```torch.load```
+        batch_dim : int
+            the index of batch dimension in the dummy_input
+        confidence: the confidence coefficient of the sparsity inference. This value is
+            actually used as the batchsize of the dummy_input.
         """
+        assert confidence > 1
         from nni.common.graph_utils import build_module_graph
-
+        # The auto inference will change the values of the parameters in the model
+        # so we need make a copy before the mask inference
+        self.ori_state_dict = copy.deepcopy(model.state_dict())
         self.bound_model = model
-        self.masks = torch.load(masks_file, map_location)
-        self.inferred_masks = dict() # key: module_name, value: ModuleMasks
-        self.dummy_input = dummy_input
-        self.torch_graph = build_module_graph(model, dummy_input)
+        self.inferred_masks = dict()  # key: module_name, value: ModuleMasks
+        self.batch_dim = batch_dim
+        self._random_model_input(dummy_input, confidence, batch_dim)
+        self.torch_graph = build_module_graph(model, self.dummy_input)
+        # dict object to save the auto inferences objects of the submodules
+        self.auto_inferences = {}
+        # the index dict to find the corresponding torch._C.Value object
+        # according to the debug name
+        # we need the dummy_input to infer the mask automaticlly, so we save
+        # the indexes from tensor's debugname to the torch._C.Value object.
+        self.debugname_to_value = {}
+        # load the mask tensor to the same device with the dummy_input
+        # self.masks save the mask tensors pruned by the user and the infered
+        # masks of the others modules
+        self.masks = torch.load(
+            masks_file, map_location if map_location is not None else str(self.device))
 
-    def infer_module_mask(self, module_name, last_module, mask=None, in_shape=None, out_shape=None):
-        """
-        Infer input shape / output shape based on the module's weight mask / input shape / output shape.
+        self.constant = {}
+        # self.internal_result save the internal output of the submodules
+        self.internal_result = {}
+        # if we enable the compilation of the sparsity, then we will modify the network
+        # architecture to resolve the sparsity conflict.
+        self.fold_bias = fold_bias
 
-        For a module:
-            Infer its input and output shape from its weight mask
-            Infer its output shape from its input shape
-            Infer its input shape from its output shape
 
-        If its input shape is changed, continue infering its predecessors
-        If its output shape is changed, continue infering its successors
+    def _random_model_input(self, dummy_input, confidence, batch_dim):
```

Comment:
suggest to return `dummy_input` and `device` then assign them to `self.dummy_input` and `self.device`. it is better than implicitly updating these two member variables within this function